mypy/util.py

```diff
@@ -158,18 +160,27 @@ def _replace(src: PathType[AnyStr], dest: PathType[AnyStr], timeout: float = 1)
     Increase wait time exponentially until total wait of timeout sec;
     on timeout, give up and reraise the last exception seen.
     """
-    n_iter = max(1, math.ceil(math.log2(timeout / 0.001)))
-    for i in range(n_iter):
-        # Last wait is ~ timeout/2, so that total wait ~ timeout.
-        wait = timeout / 2 ** (n_iter - i - 1)
+    # Find starting value for wait.
+    wait = timeout
+    total_wait: float = 0
+    while True:
+        if wait < 0.001:
+            break
+        wait /= 2
+    while True:
```

Comment:
@gvanrossum Moving the discussion here.

I could just make the time allowance in test much larger, so it will pass even though this function guarantee `timeout` only up to a factor of 2x. But this test is already slow (adds a 2-3 sec to the entire test suite) due to the file locks. 

I was trying to make this test not affect total test runtime by adding another worker but it didn't work yet (also, I fear having an extra worker might slow down tests a bit due to more process switching).